### DISCUSSION TITLE
Add a new `DType` type?

### DISCUSSION THREAD

**Discussion by sphh** [leon-thomm/Ryven#113] [Q&A]
I want to develop a new `DType`.

I know, that `DType`s are defined in `ryvencore/dtypes/dtypes.py` and that the corresponding GUI widget is selected in `ryvencore_qt/src/flows/nodes/PortItem.py`. I could add my new `DType` there, but is it also possible to add a new `DType` type without touching ryven's source code? How would I do it?

**Comment by leon-thomm** [comment]
No, currently there is no way to register new dtypes from outside. But you can simply add a custom widget instead. Something like
```python
my_widget_spec = {'widget': MyWidget, 'widget pos': 'below'}

class MyNode(Node):
    inputs = [
        NodeInputBP(label='a', add_data=my_widget_spec),
        NodeInputBP(label='b', add_data=my_widget_spec),
    ]
```

**Comment by sphh** [comment]
Yes, that's how I do it now. But it looks, like `DType`'s have more built-in features, which would be great to have.
